java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1409,9 +1409,43 @@ public ColumnVector asDate64() {
    * @return A new vector allocated on the GPU
    */
   public ColumnVector asTimestamp(TimeUnit unit) {
+    if (type == DType.STRING)
+      return asTimestamp(unit, "%Y-%m-%dT%H:%M:%SZ");
     return castTo(DType.TIMESTAMP, unit);
   }
 
+  /**
+   * Native method to parse and convert a NVString column vector to unix timestamp. A unix
+   * timestamp is a long value representing how many units since 1970-01-01 00:00:00.000 in either
+   * positive or negative direction. This mirrors the functionality spark sql's to_unix_timestamp.
+   * @param cudfColumnHandle native handle of the gdf_column being operated on.
+   * @param unit integer native ID of the time unit to parse the timestamp into.
+   * @param format strptime format specifier string of the timestamp. Used to parse and convert
+   *               the timestamp with. Supports %Y,%y,%m,%d,%H,%I,%p,%M,%S,%f,%z format specifiers.
+   * @return native handle of the resulting cudf column, used to construct the Java column vector
+   *         by the timestampToLong method.
+   */
+  private static native long stringTimestampToTimestamp(long cudfColumnHandle, int unit, String format);
+
+  /**
+   * Wrap static native string timestamp to long conversion method. Retrieves the column vector's
+   * cudf native handle and uses it to invoke the native function that calls NVStrings'
+   * timestamp2long method.
+   * @param unit time unit to parse the timestamp into.
+   * @param format strptime format specifier string of the timestamp. Used to parse and convert
+   *               the timestamp with. Supports %Y,%y,%m,%d,%H,%I,%p,%M,%S,%f,%z format specifiers.
+   * @return A new ColumnVector containing the long representations of the timestamps in the
+   *         original column vector.
+   */
+  public ColumnVector asTimestamp(TimeUnit unit, String format) {
+    assert type == DType.STRING : "A column of type string is required " +
+                                  "for .timestampToLong() operation";
+    assert unit != TimeUnit.NONE : "Time unit may not be TimeUnit.NONE";
```

Comment:
Nit: in other places in CUDF a TImeUnit.NONE really means TimeUnit.MILLISECONDS and we have updated them accordingly.  This is fine, but it might nice to stay consistent.

Reply:
I think it's reasonable to expect a user to provide a real TimeUnit, but since we're supporting constructors without the unit being specified I guess we should allow the same when casting/converting.

Should we also be supporting an asTimestamp() cast, where unit is defaulted to TimeUnit.NONE?

Reply:
Like I said this is a nit, and it really is just about consistency.  If I were designing TimeUnit there wouldn't be a NONE, but as there is and other APIs in CUDF treat it as milliseconds, it would be nice to do the same here.  I would prefer if we have an `asTimestamp()` without a TimeUnit that it defaults to milliseconds instead of NONE. Simply because not all of the C code is consistent in how it treats NONE and I think it would be less likely to cause possible problems.

Reply:
added the default cast, I think the consistency makes things a lot more intuitive and less frustrating to use.